Slim/Plugin/WiMP/ProtocolHandler.pm

```diff
@@ -315,14 +315,22 @@ sub parseDirectHeaders {
 	my ( $class, $client, $url, @headers ) = @_;
 
 	#main::DEBUGLOG && $log->is_debug && $log->debug(Data::Dump::dump(@headers));
-	my ($length, $bitrate, $ct);
+	my ($length, $rangeLength, $bitrate, $ct);
 	foreach my $header (@headers) {
 		if ( $header =~ /^Content-Length:\s*(.*)/i ) {
 			$length = $1;
 		}
 		elsif ( $header =~ /^Content-Type:\s*(\S*)/) {
 			$ct = Slim::Music::Info::mimeToType($1);
 		}
+		elsif ($header =~ m%^Content-Range:\s+bytes\s+(\d+)-(\d+)/(\d+)%i) {
+			$rangeLength = $3;
+		}
+	}
+	
+	# Content-Range: has predecence over Content-Length:
+	if ($rangeLength) {
+		$length = $rangeLength;
```

Comment:
I was wondering whether we couldn't just call 

```diff
 sub parseDirectHeaders {
        my ( $class, $client, $url, @headers ) = @_;
 
-       #main::DEBUGLOG && $log->is_debug && $log->debug(Data::Dump::dump(@headers));
-       my ($length, $bitrate, $ct);
-       foreach my $header (@headers) {
-               if ( $header =~ /^Content-Length:\s*(.*)/i ) {
-                       $length = $1;
-               }
-               elsif ( $header =~ /^Content-Type:\s*(\S*)/) {
-                       $ct = Slim::Music::Info::mimeToType($1);
-               }
-       }
+       my ($title, $bitrate, $metaint, $redir, $ct, $length, $body) = $class->SUPER::parseDirectHeaders($client, $url, @headers);
 
        $ct = 'aac' if !$ct || $ct eq 'mp4';

```

instead of parsing here?

Reply:
I think it's probably better but I needed to have a better look of the objective of that method overload. It seems that it's to get the bit rate from the header, so yes, the SUPER does the job and you can use it to fill the cache. Maybe we should wait till we are out of the wood for the rest...